pkg/cmd/cli/datamover/restore.go

```
@@ -160,7 +160,24 @@ func newdataMoverRestore(logger logrus.FieldLogger, factory client.Factory, conf
 		return nil, errors.Wrap(err, "error to create client")
 	}
 
-	cache, err := ctlcache.New(clientConfig, cacheOption)
+	var cache ctlcache.Cache
+	retry := 10
+	for {
+		cache, err = ctlcache.New(clientConfig, cacheOption)
+		if err == nil {
+			break
+		}
+
+		retry--
+		if retry == 0 {
+			break
+		}
+
+		logger.WithError(err).Warn("Failed to create client cache, need retry")
+
+		time.Sleep(time.Second)
+	}
```

Do you know why this fails? Should we use [backoffs](https://pkg.go.dev/k8s.io/client-go/util/retry#OnError)?

---

The failure is observed in Windows nodes with Calico, the pod network is not ready immediately after pod starts.
backoffs should be better. Let me change this in next PR since the same code for backupPod has already merged and there are too many things in this PR.

---

ack

---

@kaovilai
I just checked the details of the current code:
1. We don't know the detailed error that we want to retry on (it is a network error, instead of an API server error. Different network implementation may generate different errors), so we need to retry as long as `err != nil`.
2. We don't know when the network is ready, so we need to define a hard-code retry count and we don't know what retry frequency helps better. 

Because of 2, even with backoff, we have to use a backoff with `Factor` as 1 (we don't know which value is better), so exponential backoff is disabled.
Because of 1, if we use `ExponentialBackoff`, we have to return false for all errors, then the error will be swallowed and cannot be returned by `ExponentialBackoff`, since once an error is returned by the callback function `ExponentialBackoff` will quite immediately.

Therefore, I don't see any help from backoff for the current problem, to either simplify the code or improve the performance, so let's keep the code as is.

---

Thanks for the details.